Wikipedia/Categories/NSAttributedString+WMFModifyParagraphs.m

```diff
@@ -0,0 +1,25 @@
+//  Created by Monte Hurd on 8/17/15.
+//  Copyright (c) 2015 Wikimedia Foundation. Provided under MIT-style license; please copy and modify!
+
+#import "NSAttributedString+WMFModifyParagraphs.h"
+
+@implementation NSAttributedString (WMFModifyParagraphs)
+
+- (NSAttributedString*)wmf_attributedStringWithParagraphStylesAdjustments:(void (^)(NSMutableParagraphStyle* existingParagraphStyle))block {
+    if (!block) {
+        return self;
+    } else {
+        NSMutableAttributedString* mutableCopy = self.mutableCopy;
+        [self enumerateAttribute:NSParagraphStyleAttributeName
```

Comment:
I'm a bit wary of this approach, as we're always adding new constraints to a copy of the receiver, meaning you could get a buildup of attributes if this is called repeatedly.  it would be safer, i think, to recreate an "empty" attributed string w/ the receiver's `string`, then enumerate the receiver's attributes, passing them through a modifier block before setting on the new attributed string.  this way, you always have the same (or less) attributes than you started with, just with different values.